src/client/app/services/github.service.ts

```diff
@@ -119,10 +119,21 @@ export class GitHubService {
         let url = `${this._baseUrl}/gists`;
         if (!(id == null)) {
             url += `/${id}`;
-            return this._request.patch<IGist>(url, body, ResponseTypes.JSON, this._headers);
+            return this.gist(id)
+                .mergeMap(gist => {
+                    let oldFilename = Object.keys(gist.files)[0];
+                    let newFilename = Object.keys(files)[0];
+                    body.files = {};
+                    body.files[oldFilename] = {
+                        filename: newFilename,
```

Comment:
Oh, interesting.  So even when we keep the same filename, you always use the rename syntax.  Ok, I guess that makes sense that GitHub would just no-op on this sort of stuff.  Though it would still be good to add a comment that this does a rename (or no-op).